src/hooks/useModal.tsx

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import useFocusTrap from "@charlietango/use-focus-trap";
+
+export function useModal(initialState = false) {
```

Comment:
so this is fancy. do you think we could eventually pull this out into a utility? also how hard is it to add animations to this, like a fade in on open?

Reply:
Yup, we could make pull this into the component library, wouldn't be hard to add animations on demand